py-polars/polars/datatypes/classes.py

```diff
@@ -579,7 +579,15 @@ def __hash__(self) -> int:
 
     def __repr__(self) -> str:
         class_name = self.__class__.__name__
-        return f"{class_name}(categories={self.categories!r})"
+        if len(categories := self.categories) <= 6:
+            categories = [",".join(f"{cat!r}" for cat in categories)]
```

Comment:
```suggestion
            categories = [",".join(repr(cat) for cat in categories)]
```

Reply:
Not sure why you'd want to change this; the `f-string` version is more efficient (by about 25%) because it avoids the function call overhead ;)

Reply:
They are exactly the same for me. You mistyped the second example, maybe that's why?

![image](https://github.com/pola-rs/polars/assets/3502351/062ad3f6-ee2c-4dbc-bf5d-e6cb563be8bc)

`repr(x)` is definitely way more readable.

Reply:
Ahh, I did mistype the example above, and it exaggerates the degree of speedup quite a bit; the corrected example shows a more realistic ~15% speedup for me, which is more in line with what I remember from the last time I tested this in earnest (about two years back - at which point I pretty much went `f-string` only). Perhaps the degree of speedup is specific to the python version and/or architecture 🤷 
```
%timeit x = 'some_category'; y = repr(x)
# 57.4 ns ± 1.68 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

%timeit x = 'some_category'; y = f"{x!r}"
# 50.4 ns ± 0.939 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```
Anyway! This is the most micro of micro-optimisations - not a key bottleneck 🤣